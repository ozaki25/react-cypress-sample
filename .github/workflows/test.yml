name: Test
on: [pull_request]
jobs:
  unit-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: lint
        run: |
          yarn lint
      - name: test
        run: |
          yarn test
  e2e-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: e2e-test
        run: |
          yarn e2e
