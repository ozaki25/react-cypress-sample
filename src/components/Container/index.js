import React from 'react';
import styled from 'styled-components';

const Div = styled.div`
  padding: 10px 15px;
`;

function Container({ children }) {
  return <Div>{children}</Div>;
}

export default Container;
