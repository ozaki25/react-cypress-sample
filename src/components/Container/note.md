- コンテナコンポーネントです
- このコンポーネントで囲うと上下左右にpaddingがつきます