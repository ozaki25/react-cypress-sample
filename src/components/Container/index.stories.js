import React from 'react';
import Container from '.';
import note from './note.md';

export default {
  title: 'Container',
  parameters: {
    info: {
      text: note,
    },
  },
};

export const nomal = () => <Container>Hello</Container>;
