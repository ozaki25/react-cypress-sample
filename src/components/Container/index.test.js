import React from 'react';
import { snapshot } from '../../test/helper';
import Container from '.';

snapshot('Container', <Container>ボタン</Container>);
