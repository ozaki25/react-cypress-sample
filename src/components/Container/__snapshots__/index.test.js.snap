// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Container 1`] = `
.c0 {
  padding: 10px 15px;
}

<div
  className="c0"
>
  ボタン
</div>
`;
